server/backend/stream_pacer.py

```
@@ -0,0 +1,221 @@
+            yield chunk
+        return
+
+    pacer = _Pacer()
+    queue: asyncio.Queue[Any] = asyncio.Queue()
+
+    async def pump() -> None:
+        # read upstream at full speed so the stream is never slowed
+        try:
+            async for chunk in upstream:
+                await queue.put(chunk)
+        except (Exception, asyncio.CancelledError) as exc:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)